### ISSUE TITLE
Load swf cost a long time!

### ISSUE THREAD

**Issue by tidus2020** [ruffle-rs/ruffle#1996]
I want to play a swf ,10mb, it need 10sec to start. on some old android device even more than 30sec !

can this be optimized?

**Comment by ousia** [comment]
@tidus2020, many thanks for your issue report.

Unless you provide the Flash animation you have in mind, the rest of us doesn’t imagine which file you speak about 😅.

Testing may begin once you provided the file (or at least its address).

Many thanks for your help.

**Comment by tidus2020** [comment]
> @tidus2020, many thanks for your issue report.
> 
> Unless you provide the Flash animation you have in mind, the rest of us doesn’t imagine which file you speak about 😅.
> 
> Testing may begin once you provided the file (or at least its address).
> 
> Many thanks for your help.

http://main.miracle-cn.com/ruffle/lesson4.swf

**Comment by ousia** [comment]
Many thanks for the address, @tidus2020.

Not sure what might be causing it, but could you check it with latest version (I wonder whether your version included 484bd2c11a628f17c9d0b52bb44a891da32b9f46 [and this might improve things a bit]).

**Comment by ousia** [comment]
@tidus2020,

I agree that http://main.miracle-cn.com/ruffle/lesson4.swf needs longer to be loaded with Ruffle that with Flash.

I don’t know whether this can be improved. But could you provide  more information on how you use Ruffle?

1. OS and version number.
1. Processor architecture (32bit or 64bit). 
1. Browser and version number. 
1. How long does it take to open the same animation using the same browser with the Flash plugin?

I’m assuming that you are using a computer (desktop or laptop). Please, feel free correct me, if I’m wrong.

Many thanks for your help.

**Comment by Diatonator** [comment]
On 20-12-2020, desktop, it was loading for like 7 sec first time, 4-5 seconds second time, on my computer, when Flash loads it immediatly indeed.

**Comment by Diatonator** [comment]
I'll put here links to flashes on z0r.de which are loading with delay on Ruffle.

https://z0r.de/6728
https://z0r.de/3987
https://z0r.de/2015

I left the info on them in the "z0r.de issues" issue I created:
https://github.com/ruffle-rs/ruffle/issues/1766#issuecomment-744024317
https://github.com/ruffle-rs/ruffle/issues/1766#issuecomment-745517872

What they have in common is that they has "Unknown tag" errors littered in F12 console (Chrome).
From what I've read here, it could be an obfuscation processing issues? Or those flashes do really have said unknown errorneous tags and Ruffle stutters a bit on them?

**Comment by tidus2020** [comment]
I mainly in the mobile phone platform for the animation，and i do not need any as code.

**Comment by Lord-McSweeney** [comment]
This loads much faster now
